overlord/servicestate/service_control.go

```diff
@@ -129,10 +133,10 @@ func (m *ServiceManager) doServiceControl(t *state.Task, _ *tomb.Tomb) error {
 			}
 		}
 	case "restart":
-		return wrappers.RestartServices(services, nil, meter, perfTimings)
+		return wrappers.RestartServices(startupOrdered, nil, meter, perfTimings)
```

Comment:
Internally this iterates over services, so we avoid https://github.com/systemd/systemd/issues/8102 entirely which is good.